core/version_api.php

```diff
@@ -438,156 +490,70 @@ function version_remove_all( $p_project_id ) {
 	return true;
 }
 
-$g_cache_versions_project = null;
-
-/**
- * Cache version information for an array of project id's
- * @param array $p_project_id_array An array of project identifiers.
- * @return void
- */
-function version_cache_array_rows( array $p_project_id_array ) {
-	global $g_cache_versions, $g_cache_versions_project;
-
-	$c_project_id_array = array();
-
-	foreach( $p_project_id_array as $t_project_id ) {
-		if( !isset( $g_cache_versions_project[(int)$t_project_id] ) ) {
-			$c_project_id_array[] = (int)$t_project_id;
-			$g_cache_versions_project[(int)$t_project_id] = array();
-		}
-	}
-
-	if( empty( $c_project_id_array ) ) {
-		return;
-	}
-
-	$t_query = 'SELECT * FROM {project_version}
-				  WHERE project_id IN (' . implode( ',', $c_project_id_array ) . ')
-				  ORDER BY date_order DESC';
-	$t_result = db_query( $t_query );
-
-	$t_rows = array();
-	while( $t_row = db_fetch_array( $t_result ) ) {
-		$g_cache_versions[(int)$t_row['id']] = $t_row;
-
-		$t_rows[(int)$t_row['project_id']][] = $t_row['id'];
-	}
-
-	foreach( $t_rows as $t_project_id => $t_row ) {
-		$g_cache_versions_project[(int)$t_project_id] = $t_row;
-	}
-}
-
 /**
- * Return all versions for the specified project
- * @param integer $p_project_id A valid project id.
+ * Return all versions for the specified project or projects list
+ * Returned versions are ordered by reverse 'date_order'
+ * @param integer|array $p_project_ids  A valid project id, or array of ids
```

Comment:
Looks like you already support a single version or an array, then doing the same from print_api is redundant.

Reply:
Is this related to previous comment?: https://github.com/mantisbt/mantisbt/pull/1465#discussion_r257476614
about the `is_array( $p_project_ids ) ? $p_project_ids : array( $p_project_ids )`

Reply:
Yep, seems you have reasons for duplication, so I'm ok here.